opentracing-mock/src/test/java/io/opentracing/spanmanager/mdc/MDCDemo.java

```diff
@@ -0,0 +1,140 @@
+    }
+    private static final Logger logger = org.slf4j.LoggerFactory.getLogger(MDCDemo.class.getSimpleName());
+    private static final MockTracer tracer = new MockTracer(MockTracer.Propagator.PRINTER, new MDCThreadLocalSpanManager());
+
+    public void singleSpan() {
+        tracer.buildSpan("single").start().finish();
+    }
+
+    public void parentWithChild() throws Exception {
+        Span parent = tracer.buildSpan("parent").start();
+        SpanManager.VisibilityContext parentVisibility = parent.visibility().capture().on();
+        // The child will automatically know about the parent.
+        tracer.buildSpan("child")
+                .start()
+                .finish();
+        parentVisibility.off();
+        parent.finish();
+    }
+
+    public void asyncSpans() throws Exception {
+        final ExecutorService otExecutor = new TracedExecutorService(Executors.newFixedThreadPool(100), tracer.spanManager());
+
+        // Hacky lists of futures we wait for before exiting async Spans.
+        List<Future<?>> futures = new ArrayList<>();
+        final List<Future<?>> subfutures = new ArrayList<>();
+
+        // Create a parent Continuation for all of the async activity.
+        Span parent = tracer.buildSpan("asyncParent").start();
+        SpanManager.VisibilityContext parentVisibility = parent.visibility().capture().on();
+
+        // Create 10 async children.
+        for (int i = 0; i < 10; i++) {
+            final int j = i;
+            MDC.put("parent_number", String.valueOf(j));
+            futures.add(otExecutor.submit(new Runnable() {
+                @Override
+                public void run() {
+                    final Span childSpan = tracer.buildSpan("child_" + j).start();
+                    childSpan.visibility().capture().on();
+
+                    try {
+                        Thread.currentThread().sleep(1000);
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                    tracer.spanManager().active().visibility().span().log("awoke");
+                    subfutures.add(otExecutor.submit(new Runnable() {
+                        @Override
+                        public void run() {
+                            Span active = tracer.spanManager().active().visibility().span();
+                            active.log("awoke again");
+                            System.out.println(String.format("j=%d, MDC parent number: %s, MDC map: %s",
+                                    j, MDC.get("parent_number"), MDC.getCopyOfContextMap()));
+                            // Create a grandchild for each child... note that we don't *need* to use the
+                            // Continuation mechanism.
+                            Span grandchild = tracer.buildSpan("grandchild_" + j).start();
+                            grandchild.finish();
+                            childSpan.finish();
```

Comment:
Agree I wanted to keep interfaces as simple as possible (KISS). Clossable with ref count could be added to make this to look like in #115.